### ISSUE TITLE
Animation Rendering Above Fringe

### ISSUE THREAD

**Issue by scalyone** [AscensionGameDev/Intersect-Engine#430]
**Describe the bug**
I am having trouble rendering buff animations that appear above the player to render below the fringe layer. Using **Upper Layer** and **Render Below Fringe** I get my animation to appear above the player which is my desired effect however it appears above the fringe as well. When I swap to **Lower Layer** and choose **Render Above Player** I get the same exact issue. The only way I've managed to get it to render below the fringe is to have it rendered below the player which is what I'm trying to avoid. 

**Version Encountered**
0.6.2.0

**To Reproduce**
1. Go to ''Game Editors'
2. Click on 'Animation Editor'
3. In 'Upper Layer (Above Target)' click 'Render Below Fringe'
4. Create a resource by going to 'Game Editors'
5. Click on 'Resource Editor'
6. Make sure 'Below Entitites' is **not** checked.
7. Use an image that's large enough for your character sprite to walk under.
8. Save it and place it on the map.
9. Use a character and have the animation played on the character while standing under the resource.

**Expected behavior**
The expected behavior is that the animation plays under the fringe layer. It looks very odd when the player is underneath the fringe but the animation plays above the fringe.

**Screenshots**
![Animation Gif](https://s3.us-east-2.amazonaws.com/ascensiongamedev/filehost/dadad7c05b84b3b853027c98a30fe0b4.gif)

![Editor Settings](https://s3.us-east-2.amazonaws.com/ascensiongamedev/filehost/67932e2ac9b32393c97756d80751ddc0.png)

**Logs**
N/A

**Operating System:**
Both on Windows and Ubuntu

**Additional context**
N/A

**Comment by Cheshire92** [comment]
You're mentioning resources and fringe layers as one thing. But can you confirm this also actually happens when below a fringe layer?

**Comment by scalyone** [comment]
Yes the above gif looks identical when choosing 'Lower Layer (Below Target' and selecting 'Render Above Player'.

**Comment by Arufonsu** [comment]
probably not the best place for questions...
but what about animations on specific layers? i can't seems to find a way to place animations in custom layers, for example, in a layer under the ground layer!

**Comment by jcsnider** [comment]
I can confirm that the 'Render Below Fringe' option is working, as the word 'Fringe' only refers to the 2 topmost layers of Map Tiles (not resources).

So the real bug is that we have an undesired render order between player animations and resources... without excessive looping on the client this might be harder to fix, and even if we rendered the animations with each player perfectly I can foresee some other layering issues. I will have to revisit at some point when I have more time to really dig into this.

**Comment by Arufonsu** [comment]
Hello, not sure if specifically related but:

Apparently, the animations have some issues regardless the order where they are positioned whithin the editor preview and what's seeing in the game client, they basically _**differ**_ , being what i see in the editor the expected behaviour in the client as well  :'(  , hence the issue.

**Editor**
Has the expected behaviour on animations: 
they cover the previous ones when positioning downwards with respect to the Y axis (?).
![editor](https://user-images.githubusercontent.com/17498701/120610198-eda5dd80-c420-11eb-9b65-334ee6f779ff.gif)


**Client**
Doesn't properly renders the animations as expected/previously previewed, _the order feels flipped from what's seeing in the editor side._
![client](https://user-images.githubusercontent.com/17498701/120610215-f1d1fb00-c420-11eb-8ca8-0c065f9afa11.gif)

This is the latest version of the dev branch btw ~ pretty sure this issue has been around for a while anyway, even in older versions
